kubernetes/core/src/main/kotlin/com/google/kubernetes/tools/kubectl/KubectlExecutorService.kt

```diff
@@ -0,0 +1,162 @@
+                    startedProcess,
+                    commandList.joinToString(" ")
+            )
+            return kubectlProcess
+        } catch (e: Exception) {
+            val kubectlFailEventName = KUBECTL_FAIL
+
+            UsageTrackerService.getInstance().trackEvent(kubectlFailEventName)
+                    .addMetadata(
+                            METADATA_ERROR_MESSAGE_KEY, e.javaClass.name
+                    ).ping()
+            throw e
+        }
+    }
+
+    /**
+     * Check if kubectl is available, Check if the flags supplied are valid. Create and start the
+     * command line process passed in
+     */
+    fun processOutputToString(executingProcess: Process): String {
+        val reader = BufferedReader(InputStreamReader(executingProcess.inputStream))
+        val builder = StringBuilder()
+        var line = reader.readLine()
+        while (line != null) {
+            builder.append(line)
+            line = reader.readLine()
+            //so that there's no trailing or leading whitespace
+            if (line != null) builder.append(System.getProperty("line.separator"))
+        }
+        return builder.toString()
+    }
+
+    /**
+     * Check if kubectl is available, Check if the flags supplied are valid. Create and start the
+     * command line process passed in. This method waits until the process has been completed and
+     * needs to be run in a separate background thread.
+     */
+    fun startProcess(
+            executionMode: KubectlExecutorSettings.ExecutionMode,
+            configurationFlags: List<String>
+    ): String {
+
+        if (!isKubectlAvailable()) {
+            throw ExecutionException(message("kubectl.not.on.system.error"))
+        }
+
+        val executorSettings = KubectlExecutorSettings(
+                executionMode = executionMode,
+                executionFlags = configurationFlags
+        )
+
+        val executingProcess = executeKubectl(executorSettings).process
+
+        executingProcess.waitFor()
+
+        if (executingProcess.exitValue() == 0 ) {
+            return processOutputToString(executingProcess)
+        } else {
+            throw ExecutionException(message("kubectl.unknown.error"))
+        }
+    }
+}
+
+/**
+ * Data object with launched Kubectl process and its command line.
+ *
+ * @property process System process for Kubectl.
+ * @property commandLine Command line used to launch the process.
+ */
+data class KubectlProcess(val process: Process, val commandLine: String)
+
+data class KubectlExecutorSettings(
+        val executionMode: ExecutionMode,
+        val executionFlags: List<String> = ArrayList()
+) {
+
+    /** Execution mode for Kubectl, single run, continuous development, etc. */
+    enum class ExecutionMode(val modeFlag: String) {
+        CONFIG("config"),
+        CREATE("create"),
+        VERSION("version")
+    }
+
+    /** Map to check if flags are valid for an execution mode */
+    private val acceptedFlagsMap = mapOf(
```

Comment:
I think this is a part of command validation we decided to skip for this PR - could we remove this since it seems we are not using it as part of this PR?